lib/exports/colors-new/color-sets.less

```diff
@@ -0,0 +1,614 @@
+    400: hsl(28, 43%, 39%);
+}
+.set-bronze-dark() {
+    100: hsl(28, 13%, 27%);
+    200: hsl(28, 27%, 45%);
+    300: hsl(28, 58%, 67%);
+    400: hsl(28, 59%, 83%);
+}
+.set-bronze-hc() {
+    100: hsl(28, 40%, 92%);
+    200: hsl(28, 40%, 92%);
+    300: hsl(28, 43%, 65%);
+    400: hsl(28, 43%, 39%);
+}
+.set-bronze-hc-dark() {
+    100: hsl(28, 13%, 27%);
+    200: hsl(28, 13%, 27%);
+    300: hsl(28, 58%, 67%);
+    400: hsl(28, 59%, 83%);
+}
+
+// UTILITY SETS
+// border color
+.set-bc() {
+    lightest: var(--black-new-100);
+    lighter: var(--black-new-150);
+    light: var(--black-new-200);
+    medium: var(--black-new-225);
+    dark: var(--black-new-250);
+    darker: var(--black-new-300);
+}
+.set-bc-hc() {
+    lightest: var(--black-new-400);
+    lighter: var(--black-new-400);
+    light: var(--black-new-400);
+    medium: var(--black-new-400);
+    dark: var(--black-new-500);
+    darker: var(--black-new-600);
+}
+
+// background color
+.set-bg() {
+    error: var(--red-new-400);
+    danger: var(--red-new-400);
+    success: var(--green-new-400);
+    warning: var(--yellow-new-500);
+}
+
+// box-shadow
+.set-bs() {
+    sm: 0 1px 2px hsla(0, 0%, 0%, 0.05), 0 1px 4px hsla(0, 0%, 0%, 0.05), 0 2px 8px hsla(0, 0%, 0%, 0.05);
+    md: 0 1px 3px hsla(0, 0%, 0%, 0.06), 0 2px 6px hsla(0, 0%, 0%, 0.06), 0 3px 8px hsla(0, 0%, 0%, 0.09);
+    lg: 0 1px 4px hsla(0, 0%, 0%, 0.09), 0 3px 8px hsla(0, 0%, 0%, 0.09), 0 4px 13px hsla(0, 0%, 0%, 0.13);
+    xl: 0 10px 24px hsla(0, 0%, 0%, 0.05), 0 20px 48px hsla(0, 0%, 0%, 0.05), 0 1px 4px hsla(0, 0%, 0%, 0.1);
+}
+.set-bs-dark() {
+    sm: 0 1px 2px hsla(0, 0%, 0%, 0.1), 0 1px 4px hsla(0, 0%, 0%, 0.1), 0 2px 8px hsla(0, 0%, 0%, 0.1);
+    md: 0 1px 3px hsla(0, 0%, 0%, 0.11), 0 2px 6px hsla(0, 0%, 0%, 0.11), 0 3px 8px hsla(0, 0%, 0%, 0.14);
+    lg: 0 1px 4px hsla(0, 0%, 0%, 0.14), 0 3px 8px hsla(0, 0%, 0%, 0.14), 0 4px 13px hsla(0, 0%, 0%, 0.18);
+    xl: 0 10px 24px hsla(0, 0%, 0%, 0.1), 0 20px 48px hsla(0, 0%, 0%, 0.1), 0 1px 4px hsla(0, 0%, 0%, 0.15);
+}
+.set-bs-hc() {
+    sm: none;
+    md: none;
+    lg: none;
+    xl: none;
+}
+.set-bs-hc-dark() {
+    sm: none;
+    md: none;
+    lg: none;
+    xl: none;
+}
+
+// font color
+.set-fc() {
+    light: var(--black-new-400);
+    medium: var(--black-new-500);
+    dark: var(--black-new-600);
+    error: var(--red-new-400);
+    danger: var(--red-new-400);
+    success: var(--green-new-500);
+    warning: var(--yellow-new-500);
+}
+
+// focus (sets represents both light and dark mode)
+.set-focus() {
+    default: hsla(206, 100%, 40%, 0.9);
+    success: hsla(140, 40%, 75%, 0.4);
+    warning: hsla(47, 79%, 58%, 0.4);
+    error: hsla(358, 62%, 47%, 0.15);
+    muted: hsla(210, 8%, 15%, 0.1);
+}
+.set-focus-hc() {
+    default: hsla(206, 100%, 40%, 0.9);
+    success: hsla(140, 40%, 40%, 0.9);
+    warning: hsla(47, 76%, 46%, 0.9);
+    error: hsla(358, 62%, 47%, 0.9);
+    muted: hsla(210, 8%, 55%, 0.95);
+}
+
+// highlight
```

Comment:
I could not find any documentation about these highlights variables, is this something we refer only in the `code-block` and `prose` styles or also in Core or the editor for example? Try to understand if it makes sense to expose these extra variables or we could encapsulate them directly in those 2 components for the long run.

If the variables are here to be consumed outside stacks I think we should document them somewhere in our docs (maybe a small section in the Colors (new) page)

Any reason why we could not assign the variables to the new color set? Waiting confirmation from design?

Reply:
> I could not find any documentation about these highlights variables, is this something we refer only in the `code-block` and `prose` styles or also in Core or the editor for example? Try to understand if it makes sense to expose these extra variables or we could encapsulate them directly in those 2 components for the long run.

AFAICT, we only refer to these in the the `code-block` and `prose` styles. We _technically_ [redefine and reference them in Core](https://github.com/StackEng/StackOverflow/blob/main/StackOverflow.Api.V2/Content/all.css#L1008-L1100) but they all have equivalent [values](https://github.com/StackExchange/Stacks/blob/dcormier/colors-2023/lib/exports/constants-colors.less#L418-L430) and [references](https://github.com/StackExchange/Stacks/blob/dcormier/colors-2023/lib/components/code-block/code-block.less#L12-L84) in Stacks.

I think we could safely consider them private and encapsulate them in those two components. I might want to leave that for a follow up PR though just to minimize moving parts.

> Any reason why we could not assign the variables to the new color set? Waiting confirmation from design?

I left these as-is since _most_ of the values referenced are of direct color references (`hsl(…)` instead of `var(…)`). I've gone ahead and changed what I could to map to the new color palette, but many colors here are one-offs and don't map to any Stacks colors (new or legacy). I'd like to get design's opinion on this before making too many changes here, though I don't think it needs to hold up this PR.

---

@PiperLawson considering most colors used for [code blocks](https://deploy-preview-1415--stacks.netlify.app/product/components/code-blocks/#language-examples) are custom and not used elsewhere in Stacks, are we planning on updating the colors or do we want to keep those as-is?

Reply:
I also think we should not block this PR because of them. We can revisit those colors later on this month if necessary after we get feedback from design.